app/src/main/java/com/android/sample/ui/recipeOverview/RecipeOverview.kt

```diff
@@ -45,323 +39,454 @@ import androidx.compose.ui.graphics.Color
+              .fillMaxWidth()
+              .padding(padding / 2),
+      horizontalArrangement = Arrangement.SpaceBetween,
+      verticalAlignment = Alignment.CenterVertically) {
+        // Display of the preparation time
+        RecipePropertyRowText(stringResource(R.string.prep_time), "30 min", "prepTimeText")
+        // Display of the cooking time
+        RecipePropertyRowText(stringResource(R.string.cook_time), "20 min", "cookTimeText")
+        // Display of the total time that it takes
+        RecipePropertyRowText(stringResource(R.string.total_time), "50 min", "totalTimeText")
+      }
 }
 
+/**
+ * Display of the recipe properties
+ *
+ * @param title: The title of the property
+ * @param time: The time of the property
+ * @param testTag: The test tag for the property
+ */
 @Composable
-private fun RecipePropertyText(title: String, time: String, testTag: String) {
-  Column(modifier = Modifier.testTag(testTag)) {
-    Text(title, fontSize = 12.sp, color = Color.Black)
-    Spacer(modifier = Modifier.size(14.dp))
-    Text(time, fontSize = 12.sp, color = Color.Black)
-  }
+private fun RecipePropertyRowText(title: String, time: String, testTag: String) {
+  Column(
+      modifier = Modifier.testTag(testTag),
+      verticalArrangement = Arrangement.spacedBy(SMALL_PADDING.dp)) {
+        Text(
+            title,
+            style = MaterialTheme.typography.titleSmall,
+            color = MaterialTheme.colorScheme.onSecondary)
+        Text(
+            time,
+            style = MaterialTheme.typography.bodySmall,
+            color = MaterialTheme.colorScheme.onSecondary)
+      }
 }
 
-/** Display of the list of ingredients and instructions */
+/**
+ * Display of the ingredients and instructions
+ *
+ * @param ingredientsView: Boolean to determine if the ingredients are displayed
+ */
 @Composable
 private fun IngredientInstructionView(
     ingredientsView: Boolean,
-    servingsCount: Int,
-    currentRecipe: Recipe?,
-    changeServingsCount: (Int) -> Unit
+    currentRecipe: Recipe,
 ) {
-  Column {
-    if (ingredientsView) {
-      IngredientView(servingsCount, changeServingsCount, currentRecipe)
-    } else {
-      InstructionView(currentRecipe)
-    }
-  }
+
+  Column(
+      modifier = Modifier.fillMaxSize(),
+      verticalArrangement = Arrangement.spacedBy(SMALL_PADDING.dp)) {
+        if (ingredientsView) {
+          IngredientView(currentRecipe)
+        } else {
+          InstructionView(currentRecipe)
+        }
+      }
 }
 
 /** Display of the ingredients and the ability to change the number of servings */
 @Composable
-private fun IngredientView(
-    servingsCount: Int,
-    changeServingsCount: (Int) -> Unit,
-    currentRecipe: Recipe?
-) {
-  var servingsCountVar = servingsCount
+private fun IngredientView(currentRecipe: Recipe) {
+  var servingsCount by remember { mutableIntStateOf(INITIAL_NUMBER_VALUE_PER_RECIPE) }
```

Comment:
1. **Nice Modularity with `Counter`**:
   - Love that you broke out the `Counter` as a separate component! Makes it way easier to tweak just that part if anything changes with the serving adjustments.

2. **`servingsCount` Setup**:
   - Using `remember` for `servingsCount` is solid. One thought: maybe make `INITIAL_NUMBER_VALUE_PER_RECIPE` a parameter so it can be customized later if needed?

   **Quick Change**:
   ```kotlin
   private fun IngredientView(currentRecipe: Recipe, initialServings: Int = INITIAL_NUMBER_VALUE_PER_RECIPE) {
       var servingsCount by remember { mutableIntStateOf(initialServings) }
       ...
   }
   ```

3. **Good Use of `stringResource`**:
   - Nice job grabbing `servings` from `stringResource`—keeps it flexible and ready for localization!

4. **Accessibility Tip for `Counter`**:
   - Adding a `contentDescription` on `Counter` for screen readers could be handy. That way, users can hear the current servings count if they’re using accessibility features.

   **Example**:
   ```kotlin
   contentDescription = "$servingsCount servings"
   ```

Reply:
1. Thank you!
2. Good idea! I'll make that change.
3. Thanks!
4. You’re right; I added it to the modifier since Text doesn’t have a contentDescription parameter.
```kotlin
            modifier = Modifier
            .weight(1f)
            .semantics { contentDescription = "$servingsCount servings" }
```